tools/codegen/cmd/featureset-markdown.go

```diff
@@ -0,0 +1,313 @@
+package main
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"github.com/google/go-cmp/cmp"
+	"github.com/openshift/api/tools/codegen/pkg/utils"
+	"github.com/spf13/pflag"
+	"io"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/util/sets"
+	"os"
+	"path/filepath"
+	kyaml "sigs.k8s.io/yaml"
+	"sort"
+	"strings"
+	"time"
+
+	"github.com/spf13/cobra"
+)
+
+type FeatureSetOptions struct {
+	In     io.Reader
+	Out    io.Writer
+	ErrOut io.Writer
+
+	FeatureSetManifestDir string
+	OutputFile            string
+	Verify                bool
+}
+
+func NewFeatureSetOptions(in io.Reader, out, errOut io.Writer) *FeatureSetOptions {
+	return &FeatureSetOptions{
+		In:                    in,
+		Out:                   out,
+		ErrOut:                errOut,
+		FeatureSetManifestDir: filepath.Join("payload-manifests", "featuregates"),
+		OutputFile:            "features.md",
+	}
+}
+
+func NewFeatureSetFlagsCommand(in io.Reader, out, errOut io.Writer) *cobra.Command {
+	o := NewFeatureSetOptions(in, out, errOut)
+
+	cmd := &cobra.Command{
+		Use:   "featureset-markdown",
+		Short: "featureset-markdown generates a markdown document summarizing current featuregate status.",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			ctx, cancelFn := context.WithTimeout(context.Background(), 10*time.Second)
+			defer cancelFn()
+
+			if err := o.Validate(); err != nil {
+				return err
+			}
+
+			return o.Run(ctx)
+		},
+	}
+	o.AddFlags(cmd.Flags())
+
+	return cmd
+}
+
+func (o *FeatureSetOptions) Validate() error {
+	if len(o.FeatureSetManifestDir) == 0 {
+		return fmt.Errorf("--featureset-manifest-path is required")
+	}
+	if _, err := os.ReadDir(o.FeatureSetManifestDir); err != nil {
+		return fmt.Errorf("--featureset-manifest-path cannot be read: %w", err)
+	}
+	if len(o.OutputFile) == 0 {
+		return fmt.Errorf("--output-file is required")
+	}
+
+	return nil
+}
+
+func (o *FeatureSetOptions) AddFlags(flags *pflag.FlagSet) {
+	flags.StringVar(&o.FeatureSetManifestDir, "featureset-manifest-path", o.FeatureSetManifestDir, "path to directory containing the FeatureGate YAMLs for each FeatureSet,ClusterProfile tuple.")
+	flags.StringVar(&o.OutputFile, "output-file", o.OutputFile, "path to markdown file detailing FeatureGates.")
+	flags.BoolVar(&o.Verify, "verify", o.Verify, "Verify the content has not changed.")
```

Comment:
Might conflict with the global verify flag?

Reply:
> Might conflict with the global verify flag?

doesn't seem to in practice.